packages/runtime/runtime-definitions/src/dataStoreContext.ts

```diff
@@ -184,13 +209,20 @@ export interface IFluidDataStoreChannel extends
      * Called to bind the runtime to the container.
      * If the container is not attached to storage, then this would also be unknown to other clients.
      */
-     bindToContext(): void;
+    bindToContext(): void;
 
     /**
+     * @deprecated - This will be removed in favor of makeVisibleAndAttachGraph.
      * Runs through the graph and attaches the bound handles. Then binds this runtime to the container.
      */
     attachGraph(): void;
 
+    /**
+     * Makes the data store channel visible in the container. Also, runs through its graph and attaches all
+     * bound handles that represent its dependencies in the container's graph.
+     */
+    makeVisibleAndAttachGraph?(): void;
```

Comment:
It's really hard to follow all attachment code :)
I think one reason - it's hard to think about directionality of calls. I.e. I'd think that most binding flows (including creation flow) would have direction channel -> context and ideally would not need context -> channel communication. Container attachment flow needs context -> channel communication.

This API seems like includes both. Glancing at code, I think it's only needed for root data stores, and in future may be removed (I think, assuming we remove attachment flow in its current form).
Is that correct assessment?

Reply:
Yes, you are right. Making data store locally visible (binding) flow always happens from channel -> context. Making them globally visible (attaching) happens from context -> channel.
For root-data stores too, the making locally visible flow is the same where the container runtime calls the channel which then calls context. So, channel -> context is preserved. However, there is an additional call from the context layer to channel layer first.

Another way to remove this will be to make the alias map behave like a DDS. When aliasing a data store, store its handle in the alias map which calls attachGraph() on it. This will make the behavior similar to non-root data stores.